plugins/backgroundrunner.koplugin/commandrunner.lua

```diff
@@ -0,0 +1,62 @@
+local logger = require("logger")
+
+local CommandRunner = {
+    pio = nil,
+    job = nil,
+}
+
+function CommandRunner:start(job)
+    assert(self ~= nil)
+    assert(self.pio == nil)
+    assert(self.job == nil)
+    self.job = job
+    self.job.start_sec = os.time()
+    assert(type(self.job.executable) == "string")
+    self.pio = io.popen("sh plugins/backgroundrunner.koplugin/luawrapper.sh " ..
+                        "\"" .. self.job.executable .. "\"")
+end
+
+--- Polls the status of self.pio.
+-- @return a table contains the result from luawrapper.sh. Returns nil if the
+--         command has not been finished.
+function CommandRunner:poll()
+    assert(self ~= nil)
+    assert(self.pio ~= nil)
+    assert(self.job ~= nil)
+    local line = self.pio:read()
+    if line == "" then
+        return nil
+    else
+        if line == nil then
+            -- The binary crashes without output. This should not happen.
+            self.job.result = 223
```

Comment:
I simply try to avoid a common error code from sh. But I have a very limited understanding to the sh, so feel free to suggest any other values.

P.S. Android does not have a bash AFAIK, so I use sh here.